### ISSUE TITLE
Frontier API module parsing throws "KeyNotFound" exceptions for modules not included in EDDI data definitions

### ISSUE THREAD

**Issue by Tkael** [EDCD/EDDI#715]
## EDDI version in which issue found
3.0.1-rc2

## Steps to reproduce

Dock at stations with the Frontier API active and modules not yet defined in EDDI's data definitions (such as Krait or challenger armor, if #713 hasn't yet been incorporated).

## Expected
No exceptions

## Observed
(repeated for each unrecognized module)
```cs
2018-06-29T02:29:12 [Error] EDDI:conditionallyRefreshProfile Exception obtaining profile System.Collections.Generic.Dictionary`2[System.String,System.Object]
2018-06-29T02:29:13 [Error] EDDI:conditionallyRefreshProfile Exception obtaining profile System.Collections.Generic.KeyNotFoundException: The given key was not present in the dictionary.
   at System.Collections.Generic.Dictionary`2.get_Item(TKey key)
   at EddiDataDefinitions.Module.FromEliteID(Int64 eliteID) in EDDI\DataDefinitions\ModuleDefinitions.cs:line 1100
   at EddiCompanionAppService.CompanionAppService.OutfittingFromProfile(Object json) in EDDI\CompanionAppService\CompanionAppService.cs:line 623
   at EddiCompanionAppService.CompanionAppService.Station(String systemName) in EDDI\CompanionAppService\CompanionAppService.cs:line 303
   at Eddi.EDDI.conditionallyRefreshProfile() in EDDI\EDDI\EDDI.cs:line 1794
```

## Investigation
Improved null handling + a fall back to lookup by the module's edname are needed.

**Comment by Tkael** [comment]
Submitted PR #741.

**Comment by richardbuckle** [comment]
Fixed for 3.0.2-b1.
